bookkeeper-server/src/main/java/org/apache/bookkeeper/bookie/GarbageCollectorThread.java

```diff
@@ -547,11 +585,14 @@ protected void compactEntryLog(EntryLogMetadata entryLogMeta) {
             LOG.info("Extracting entry log meta from entryLogId: {}", entryLogId);
 
             try {
-                // Read through the entry log file and extract the entry log meta
+                // Read through the entry log file and extract the entry log
+                // meta
                 EntryLogMetadata entryLogMeta = entryLogger.getEntryLogMetadata(entryLogId);
-                removeIfLedgerNotExists(entryLogMeta);
+                removeIfLedgerNotExists(entryLogId, entryLogMeta, false);
                 if (entryLogMeta.isEmpty()) {
                     entryLogger.removeEntryLog(entryLogId);
+                    // remove it from entrylogmetadata-map if it presents into map
+                    removeEntryLogRecordSafely(entryLogId);
```

Comment:
I don't understand why you want to swallow the exception here.

Reply:
I think we can avoid failing entire gc-task due to single record failure and gc should continue by ignoring that record because that will not cause any harm except it will not remove that specific entry/ledger in that gc-run.

Reply:
Did you ever see this case in production?
If we do not have a strong motivation O prefer  to leave the code as before here